pkg/controllermanager/controller/shoot/maintenance/reconciler_test.go

```diff
@@ -1269,9 +1274,192 @@ var _ = Describe("Shoot Maintenance", func() {
 			Expect(shoot.Spec.Provider.Workers[1].Maximum).To(Equal(int32(2)))
 		})
 	})
+
+	Describe("#MaintainFeatureGatesForShoot", func() {
```

Comment:
Is it possible to add integration test (envtest) test cases in `test/integration/controllermanager/shoot/maintenance` for the new scenarios that are introduced with this PR?